src/libraries/Microsoft.PowerFx.Core/Types/DType.cs

```diff
@@ -3196,56 +3330,142 @@ public virtual bool CoercesTo(DType typeDest, out bool isSafe, out DType coercio
                     break;
                 case DKind.Currency:
                     // Ill-formatted strings coerce to null; unsafe.
                     isSafe = Kind != DKind.String;
                     doesCoerce = Kind == DKind.String ||
                                  Kind == DKind.Number ||
-                                 Boolean.Accepts(this);
+                                 Kind == DKind.Decimal ||
+                                 Boolean.Accepts(this, exact: true, useLegacyDateTimeAccepts: false, usePowerFxV1CompatibilityRules: usePowerFxV1CompatibilityRules);
                     break;
-                case DKind.Decimal:
+                case DKind.Decimal:
                     // Ill-formatted strings coerce to null; unsafe.
                     isSafe = Kind != DKind.String && Kind != DKind.Number;
                     doesCoerce = Kind == DKind.String ||
-                                 Number.Accepts(this) ||
-                                 Boolean.Accepts(this) ||
-                                 Decimal.Accepts(this) ||
-                                 DateTime.Accepts(this);
+                                 Currency.Accepts(this, exact: true, useLegacyDateTimeAccepts: false, usePowerFxV1CompatibilityRules: usePowerFxV1CompatibilityRules) ||
+                                 Number.Accepts(this, exact: true, useLegacyDateTimeAccepts: false, usePowerFxV1CompatibilityRules: usePowerFxV1CompatibilityRules) ||
+                                 Boolean.Accepts(this, exact: true, useLegacyDateTimeAccepts: false, usePowerFxV1CompatibilityRules: usePowerFxV1CompatibilityRules) ||
+                                 Decimal.Accepts(this, exact: true, useLegacyDateTimeAccepts: false, usePowerFxV1CompatibilityRules: usePowerFxV1CompatibilityRules) ||
+                                 DateTime.Accepts(this, exact: true, useLegacyDateTimeAccepts: false, usePowerFxV1CompatibilityRules: usePowerFxV1CompatibilityRules) ||
+                                 Date.Accepts(this, exact: true, useLegacyDateTimeAccepts: false, usePowerFxV1CompatibilityRules: usePowerFxV1CompatibilityRules) ||
+                                 Time.Accepts(this, exact: true, useLegacyDateTimeAccepts: false, usePowerFxV1CompatibilityRules: usePowerFxV1CompatibilityRules);
                     break;
                 case DKind.String:
                     doesCoerce = Kind != DKind.Color && Kind != DKind.Control && Kind != DKind.DataEntity && Kind != DKind.OptionSet && Kind != DKind.View && Kind != DKind.Polymorphic && Kind != DKind.File && Kind != DKind.LargeImage;
+                    break;
+                case DKind.Guid:
+                    if (usePowerFxV1CompatibilityRules)
+                    {
+                        doesCoerce = String.Accepts(this, exact: true, useLegacyDateTimeAccepts: false, usePowerFxV1CompatibilityRules: true);
```

Comment:
>String.Accepts(this, exact: true, useLegacyDateTimeAccepts: false, usePowerFxV1CompatibilityRules: true); [](http://example.com/codeflow?start=37&length=105)

Is string to GUID coercion supported?

Reply:
Yes, we need to have a coercion relationship now. Before it was an accepts relationship, now we need to add a coercion node.